hal/armv7a/zynq7000/zynq.c

```diff
@@ -91,6 +90,331 @@ static void _zynq_slcrUnlock(void)
 }
 
 
+static int _zynq_setAmbaClk(u32 dev, u32 state)
+{
+	/* Check max dev position in amba register */
+	if (dev > 24)
+		return -1;
+
+	_zynq_slcrUnlock();
+	*(zynq_common.slcr + slcr_aper_clk_ctrl) = (*(zynq_common.slcr + slcr_aper_clk_ctrl) & ~(1 << dev)) | (!!state << dev);
+	_zynq_slcrLock();
+
+	return 0;
+}
+
+
+static int _zynq_getAmbaClk(u32 dev, u32 *state)
+{
+	/* Check max dev position in amba register */
+	if (dev > 24)
+		return -1;
+
+	_zynq_slcrUnlock();
+	*state = (*(zynq_common.slcr + slcr_aper_clk_ctrl) >> dev) & 0x1;
+	_zynq_slcrLock();
+
+	return 0;
+}
+
+
+static int _zynq_setDevClk(int dev, char divisor0, char divisor1, char srcsel, char clkact0, char clkact1)
+{
+	u32 id = 0;
+
+	_zynq_slcrUnlock();
+	switch (dev) {
+		case pctl_ctrl_usb0_clk:
+		case pctl_ctrl_usb1_clk:
+			id = dev - pctl_ctrl_usb0_clk;
+			*(zynq_common.slcr + slcr_usb0_clk_ctrl + id) = (*(zynq_common.slcr + pctl_ctrl_usb0_clk + id) & ~0x00000070) | (srcsel & 0x7) << 4;
+			break;
```

Comment:
What do you think of this solution:
```suggestion
	volatile u32 *p;
	u32 tmp1, tmp2;

		case pctl_ctrl_usb0_clk:
		case pctl_ctrl_usb1_clk:
			id = dev - pctl_ctrl_usb0_clk;
			*p = zynq_common.slcr + slcr_usb0_clk_ctrl + id;
			tmp1 = *p & ~0x00000070;
			tmp2 = (srcsel & 0x7) << 4;
			break;
```
and after the `switch` statement
```
	*p = tmp1 | tmp2;
```

Reply:
I don't want to add extra variables. I am leaving it as it is.